Updating the K-NN Filters documentation due to recent enhancements in… (#4987)
opensearch-project/documentation-website: _search-plugins/knn/filter-search-knn.md

- [Efficient k-NN filtering](#efficient-k-nn-filtering): This approach applies filtering _during_ the k-NN search, as opposed to before or after the k-NN search, which ensures that `k` results are returned (if there are at least `k` results in total). This approach is supported by the following engines:
  - Lucene engine with a Hierarchical Navigable Small World (HNSW) algorithm (k-NN plugin versions 2.4 and later) 

>   - Faiss engine with an HNSW algorithm (k-NN plugin versions 2.9 or later) or IVF algorithm (k-NN plugin versions 2.10 or later)

Reviewer: ```suggestion
  - Faiss engine with an HNSW algorithm (k-NN plugin versions 2.9 and later) or IVF algorithm (k-NN plugin versions 2.10 and later)
```

>   - Faiss engine with an HNSW algorithm (k-NN plugin versions 2.9 or later) or IVF algorithm (k-NN plugin versions 2.10 or later)

Reviewer: Does IVF need to be defined?

Author: Yes.

### Faiss k-NN filter implementation 

> For k-NN searches, you can use `faiss` filters with an HNSW algorithm (k-NN plugin versions 2.9 or later) or IVF algorithm (k-NN plugin versions 2.10 or later).

Reviewer: ```suggestion
For k-NN searches, you can use `faiss` filters with an HNSW algorithm (k-NN plugin versions 2.9 and later) or IVF algorithm (k-NN plugin versions 2.10 and later).
```

When you specify a Faiss filter for a k-NN search, the Faiss algorithm decides whether to perform an exact k-NN search with pre-filtering or an approximate search with modified post-filtering. The algorithm uses the following variables:

- N: The number of documents in the index.
- P: The number of documents in the document subset after the filter is applied (P <= N).
- k: The maximum number of vectors to return in the response.

> - R: The number of results returned after doing the Filtered Approximate Nearest Neighbor Search.

Reviewer: ```suggestion
- R: The number of results returned after performing the filtered approximate nearest neighbor search.
```

- FilterThreshold: A per index level threshold defined via `knn.advanced.filtered_exact_search_threshold` setting( [Ref]({{site.url}}{{site.baseurl}}/search-plugins/knn/settings/)) to switch to exact search. (P <= FilterThreshold)

> - MaxDistanceComputations: The maximum number of distance computations allowed in exact search if FilterThreshold is not set. This value cannot be changed. 

Reviewer 2: ```suggestion
- max distance computations: The maximum number of distance computations allowed in exact search if f filtered threshold is not set. This value cannot be changed. 
```

### Faiss k-NN filter implementation 

For k-NN searches, you can use `faiss` filters with an HNSW algorithm (k-NN plugin versions 2.9 or later) or IVF algorithm (k-NN plugin versions 2.10 or later).

When you specify a Faiss filter for a k-NN search, the Faiss algorithm decides whether to perform an exact k-NN search with pre-filtering or an approximate search with modified post-filtering. The algorithm uses the following variables:

- N: The number of documents in the index.
- P: The number of documents in the document subset after the filter is applied (P <= N).
- k: The maximum number of vectors to return in the response.
- R: The number of results returned after doing the Filtered Approximate Nearest Neighbor Search.

> - FT: An index-level threshold defined in the [`knn.advanced.filtered_exact_search_threshold` setting]({{site.url}}{{site.baseurl}}/search-plugins/knn/settings/) that specifies to switch to exact search.

Reviewer 2: ```suggestion
- FT (filtered threshold): An index-level threshold defined in the [`knn.advanced.filtered_exact_search_threshold` setting]({{site.url}}{{site.baseurl}}/search-plugins/knn/settings/) that specifies to switch to exact search.
```

> - MDC: The maximum number of distance computations allowed in exact search if `f` filtered threshold is not set. This value cannot be changed.

Reviewer 2: ```suggestion
- MDC (max distance computations): The maximum number of distance computations allowed in exact search if `FT` (filtered threshold) is not set. This value cannot be changed.
```
